### DISCUSSION TITLE
GraphQL API v4 - Get or search for topics list

### DISCUSSION THREAD

**Discussion by callain** [community/community#24806] [Apps, API and Webhooks]
<p>Hello,</p>
<p>Is it possible to fetch all topics or all features topics or just search for topics (featured if possible) with the GraphQL API ?</p>
<p>Should I consider using the v3 API and if yes, do you have any date where it will be safe for production ?</p>
<p>I’m looking for a way to add Topic Suggestion or searching topics in my app.</p>
<p>Thank you</p>

**Comment by lee-dohm** [comment]
<p>No, it isn’t currently possible to get a list of all topics, featured topics, or search for topics via GraphQL (or, to my knowledge, with REST for that matter). You can search for repositories with a given topic, however.</p>
<p>I hope that helps!</p>

**Comment by callain** [comment]
<p>That’s unfortunate, I guess I will hardcode a list of topics in my application.</p>
<p>Can you create a feature request for this ?</p>

**Comment by AndreaGriffiths11** [comment, marked as answer]
<p>Thanks for the feedback! I’ve taken your suggestion and passed it along to the appropriate teams. Thanks again for reaching out :grinning:</p>

**Reply by joeyfigaro** [reply]
Is there anywhere we can track or vote for this feature request? Still need to be able to do this a few years later.

**Comment by atmosx** [comment]
<p>How can you do that GraphQL?</p>

**Comment by shaejaz** [comment]
Any updates on this feature?

The REST API supports searching for topics as documented here: https://docs.github.com/en/rest/search?apiVersion=2022-11-28#search-topics

Would be nice to get this in the GraphQL API as well. Unless I'm missing something and it's somewhere else in the Explorer:

![image](https://user-images.githubusercontent.com/29524828/216960194-b9a84808-ca04-4d23-90ab-b8ddfa7330a9.png)
